test/pummel/test-string-decoder-large-buffer.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../common');
+
+const assert = require('assert');
+const { StringDecoder } = require('node:string_decoder');
+const kStringMaxLength = require('buffer').constants.MAX_STRING_LENGTH;
+
+const size = 2 ** 31;
```

Comment:
Would it be better to set `size` to `MAX_STRING_LENGTH`+1 ?

Reply:
> Would it be better to set `size` to `MAX_STRING_LENGTH`+1 ?

There is a integer overflow when decoding a large buffer with size exceeds maximum integer. so I think  `MAX_STRING_LENGTH`+1(< maximum integer value) is covered